spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicSameContainerFactoryIntegrationTests.java

```diff
@@ -121,41 +118,52 @@ static class RetryableKafkaListener {
 		public void listen(String in, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
 			countDownLatchContainer.countDownLatch1.countDown();
 			logger.warn(in + " from " + topic);
-			throw new RuntimeException("test");
+			throw new RuntimeException("from RetryableKafkaListener");
 		}
 
 		@DltHandler
 		public void dlt(String in, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
-			countDownLatchContainer.countDownLatchDltOne.countDown();
+			countDownLatchContainer.countDownLatchDlt1.countDown();
 			logger.warn(in + " from " + topic);
 		}
 	}
 
 	@Component
-	static class BasicKafkaListener {
+	static class SecondRetryableKafkaListener {
 
 		@Autowired
 		CountDownLatchContainer countDownLatchContainer;
 
+		@RetryableTopic
 		@KafkaListener(topics = RetryTopicSameContainerFactoryIntegrationTests.SECOND_TOPIC)
 		public void listen(String in, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
-			logger.info(in + " from " + topic);
-			throw new RuntimeException("another test");
+			countDownLatchContainer.countDownLatch2.countDown();
+			logger.warn(in + " from " + topic);
+			throw new RuntimeException("from SecondRetryableKafkaListener");
+		}
+
+
+		@DltHandler
+		public void dlt(String in, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+			countDownLatchContainer.countDownLatchDlt2.countDown();
+			logger.warn(in + " from " + topic);
 		}
 	}
 
 	@Component
 	static class CountDownLatchContainer {
 
 		CountDownLatch countDownLatch1 = new CountDownLatch(4);
-		CountDownLatch countDownLatch2 = new CountDownLatch(1);
-		CountDownLatch countDownLatchDltOne = new CountDownLatch(1);
-		CountDownLatch customizerLatch = new CountDownLatch(6);
+		CountDownLatch countDownLatch2 = new CountDownLatch(3);
+
+		CountDownLatch countDownLatchDlt1 = new CountDownLatch(1);
+		CountDownLatch countDownLatchDlt2 = new CountDownLatch(1);
+		CountDownLatch customizerLatch = new CountDownLatch(9);
 	}
 
 	@EnableKafka
 	@Configuration
-	static class Config extends RetryTopicConfigurationSupport {
+	static class Config {
```

Comment:
We shouldn't extend `RetryTopicConfigurationSupport` and shouldn't inject our`ThreadPoolTaskScheduler` own `https://github.com/spring-projects/spring-kafka/pull/2555/files#diff-80d8fe2567e6640c6d1edada33ec0bf783012606c2264cf4fc6b3aed4cca9742L242`